src/prefect/agent.py

```diff
@@ -177,6 +189,85 @@ async def get_and_submit_flow_runs(self) -> List[FlowRun]:
 
         return submittable_runs
 
+    async def check_for_cancelled_flow_runs(self):
+        if not self.started:
+            raise RuntimeError("Agent is not started. Use `async with OrionAgent()...`")
+
+        self.logger.debug("Checking for cancelled flow runs...")
+
+        work_queue_names = set()
+        async for work_queue in self.get_work_queues():
+            work_queue_names.add(work_queue.name)
+
+        cancelling_flow_runs = await self.client.read_flow_runs(
+            flow_run_filter=FlowRunFilter(
+                state=FlowRunFilterState(
+                    name=FlowRunFilterStateName(any_=["Cancelling"])
+                ),
+                work_queue_name=FlowRunFilterWorkQueueName(any_=list(work_queue_names)),
+            ),
+        )
+
+        if cancelling_flow_runs:
+            self.logger.info(
+                f"Found {len(cancelling_flow_runs)} flow runs awaiting cancellation."
+            )
+
+        async with anyio.create_task_group() as tg:
+            for flow_run in cancelling_flow_runs:
+
+                # Avoid duplicate cancellation calls
+                if flow_run.id in self.cancelling_flow_run_ids:
+                    continue
+                else:
+                    self.cancelling_flow_run_ids.add(flow_run.id)
+
+                tg.start_soon(self.cancel_run, flow_run)
+
```

Comment:
Purely a stylistic thing - is an explicit `if <whatever>: continue; else: <do stuff>` preferred over something like

```python
if flow_run.id not in self.cancelling_flow_run_ids:
    self.cancelling_flow_run_ids.add(flow_run.id)
    tg.start_soon(self.cancel_run, flow_run)
```

Reply:
Yeah this continue else clause is pretty silly. Usually I'll just exclude the `else` here...

```python
                # Avoid duplicate cancellation calls
                if flow_run.id in self.cancelling_flow_run_ids:
                    continue

                self.cancelling_flow_run_ids.add(flow_run.id)
                tg.start_soon(self.cancel_run, flow_run)
```

I think I prefer that because it reduces nesting and makes the comment really clear